pkg/control/control.go

```diff
@@ -0,0 +1,23 @@
+package control
+
+import (
+	"context"
+	"fmt"
+
+	orgv1 "github.com/appuio/control-api/apis/organization/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func GetSalesOrder(ctx context.Context, k8sClient client.Client, orgId string) (string, error) {
+
+	org := &orgv1.Organization{}
+	err := k8sClient.Get(ctx, client.ObjectKey{Name: orgId}, org)
+	if err != nil {
+		return "", fmt.Errorf("cannot get Organization object '%s', err: %v", orgId, err)
+	}
+	if org.Status.SaleOrderName != "" {
```

Comment:
IMHO this makes is a bit easier to read and you don't actually need an else branch.

This way the happy path ends with the function.

```suggestion
	if org.Status.SaleOrderName == "" {
    return "", fmt.Errorf("Cannot get SalesOrder from organization object '%s', err: %v", orgId, err)
  }
  return org.Status.SaleOrderName, nil
```